sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/SetLocaleForDataTypes.cs

```diff
@@ -0,0 +1,213 @@
+                                    typeName));
+                            }
+                        });
+                });
+        }
+
+        private static SyntaxNode GetAssignmentTargetVariable(ObjectCreationExpressionSyntax objectCreation)
+        {
+            var parent = objectCreation.GetSelfOrTopParenthesizedExpression().Parent;
+
+            var assignment = parent as AssignmentExpressionSyntax;
+            if (assignment != null)
+            {
+                return assignment.Left;
+            }
+
+            var equalsClause = parent as EqualsValueClauseSyntax;
+            if (equalsClause != null)
+            {
+                var variableDeclaration = equalsClause.Parent.Parent as VariableDeclarationSyntax;
+                if (variableDeclaration != null)
+                {
+                    return variableDeclaration.Variables.Last();
+                }
+            }
+
+            return null;
+        }
+
+        private static IPropertySymbol GetPropertySymbol(AssignmentExpressionSyntax assignment, SemanticModel model)
+        {
+            var conditionalAccess = assignment.Left as ConditionalAccessExpressionSyntax;
+
+            var syntax = conditionalAccess != null
+                ? conditionalAccess.WhenNotNull
+                : assignment.Left;
+
+            return model.GetSymbolInfo(syntax).Symbol as IPropertySymbol;
+        }
+
+        private static ISymbol GetAccessedVariable(AssignmentExpressionSyntax assignment, SemanticModel model)
+        {
+            var variable = assignment.Left.RemoveParentheses();
+
+            var identifier = variable as IdentifierNameSyntax;
+            if (identifier != null)
+            {
+                var leftSideOfParentAssignment = identifier
+                    .FirstAncestorOrSelf<ObjectCreationExpressionSyntax>()
+                    ?.FirstAncestorOrSelf<AssignmentExpressionSyntax>()
+                    ?.Left;
+                if (leftSideOfParentAssignment != null)
+                {
+                    return model.GetSymbolInfo(leftSideOfParentAssignment).Symbol;
+                }
+
+                var lastVariable = identifier.FirstAncestorOrSelf<VariableDeclarationSyntax>()?.Variables.LastOrDefault();
+                return lastVariable != null
+                    ? model.GetDeclaredSymbol(lastVariable)
+                    : null;
+            }
+
+            var memberAccess = variable as MemberAccessExpressionSyntax;
+            if (memberAccess != null)
+            {
+                return memberAccess.Expression != null
+                    ? model.GetSymbolInfo(memberAccess.Expression).Symbol
+                    : null;
+            }
+
+            var conditionalAccess = variable as ConditionalAccessExpressionSyntax;
+            return conditionalAccess?.Expression == null
+                ? model.GetSymbolInfo(conditionalAccess.Expression).Symbol
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') 'conditionalAccess' is null on at least one execution path. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS2259)